test/correctness/constraints.cpp

```diff
@@ -62,7 +62,7 @@ int main(int argc, char **argv) {
     error_occurred = false;
     h.realize(image1);
     // Also check it compiles ok without an inferred argument list
-    h.compile_to_assembly("h.s", Internal::vec<Argument>(image1), "h");
+    h.compile_to_assembly("h.s", Internal::vec<Argument>(Buffer(image1).as_input()), "h");
```

Comment:
OK, I'll back out the change to Buffer, and have operator Argument always imply an InputBuffer.